samples/qir/oracle-generator/qsharp/Program.qs

```diff
@@ -0,0 +1,50 @@
+namespace OracleCompiler {
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Measurement;
+
+    operation Majority3(inputs : (Qubit, Qubit, Qubit), output : Qubit) : Unit {
+        // The implementation of this operation will be
+        // automatically derived from the description in
+        // OracleCompiler.Classical.Majority3
+    }
+
+    @EntryPoint()
+    operation RunProgram() : Unit {
+        InitOracleCompilerFor(OracleCompiler.Classical.Majority3);
+
+        use (a, b, c) = (Qubit(), Qubit(), Qubit());
+        use f = Qubit();
+
+        for ca in [false, true] {
+            for cb in [false, true] {
+                for cc in [false, true] {
+                    within {
+                        if ca { X(a); }
+                        if cb { X(b); }
+                        if cc { X(c); }
+                    } apply {
+                        Majority3((a, b, c), f);
+                        let result = IsResultOne(MResetZ(f));
+
+                        Message($"{cc} {cb} {ca} -> {result}");
+                    }
+                }
+            }
+        }
+    }
+
+    internal function InitOracleCompilerFor<'In, 'Out>(func : 'In -> 'Out) : Unit {
```

Comment:
What is this function used for?

Reply:
I'll add a comment. But for now it's actually only used as a workaround to avoid that all the operations and functions referred to in this function are included into the generated QIR code. If they are not used, they are going to be stripped.

Reply:
Makes sense, I figured it was something like that. As you say, adding a comment can help the reader understand why that workaround is needed.